cdc: add more description about Unified Sorter (#5093)
pingcap/docs: ticdc/manage-ticdc.md

- `--start-ts`: Specifies the starting TSO of the `changefeed`. From this TSO, the TiCDC cluster starts pulling data. The default value is the current time.
- `--target-ts`: Specifies the ending TSO of the `changefeed`. To this TSO, the TiCDC cluster stops pulling data. The default value is empty, which means that TiCDC does not automatically stop pulling data.
- `--sort-engine`: Specifies the sorting engine for the `changefeed`. Because TiDB and TiKV adopt distributed architectures, TiCDC must sort the data changes before writing them to the sink. This option supports `memory`/`unified`/`file`.

    - `memory`: Sorts data changes in memory.
    - `unified`: When `unified` is used, TiCDC prefers data sorting in memory. If the memory is insufficient, TiCDC automatically uses the disk to store the temporary data. This is the default value of `--sort-engine`.

>     - `file`: Entirely uses the disk to store the temporary data. This feature is **deprecated**. It is **NOT recommended** to use it in **any** situation.

Reviewer: ```suggestion
    - `file`: Entirely uses the disk to store the temporary data. This option is **deprecated**. It is **NOT recommended** to use it in **any** situation.
```

Author: `--sort-engine` is an option. `memory`,`unified`, and `file` are optional values for `--sort-engine`. On L105, it says "This option supports `memory`/`unified`/`file`". Will it be confusing if we also use the word "option" here?

Reviewer: Yes, you're right. Ignore this comment.

btw. — if it's no recommended to use `file` in **any situation**, why document it at all?

Author: @liuzix PTAL.

Reviewer 2: > Yes, you're right. Ignore this comment.
> 
> btw. — if it's not recommended to use `file` in **any situation**, why document it at all?

There were users who tried to use `file`. The documentation is designed to explicitly warn about this.

> **Warning:**
>
> For tables without a valid index, operations such as `INSERT` and `REPLACE` are not reentrant, so there is a risk of data redundancy. TiCDC guarantees that data is distributed only at least once during the replication process. Therefore, enabling this feature to replicate tables without a valid index will definitely cause data redundancy. If you do not accept data redundancy, it is recommended to add an effective index, such as adding a primary key column with the `AUTO RANDOM` attribute.

## Unified Sorter

Unified sorter is the sorting engine in TiCDC. This feature is enabled by default. It can mitigate OOM problems caused by the following scenarios:

> + The data changes subscription task in TiCDC is paused for a long time, during which a large amount of incremental data is accumulated to be replicated.

Reviewer 2: ```suggestion
+ The data replication task in TiCDC is paused for a long time, during which a large amount of incremental data is accumulated and need to be replicated.
```
`accumulated to be replicated` sounds unnatural to me, because it sounds purposive and implies that TiCDC or some other component actively collects the data for the purpose of replication, whereas in fact TiCDC does nothing while paused.

> + The data replication task in TiCDC is paused for a long time, during which a large amount of incremental data is accumulated and need to be replicated.

Reviewer 2: ```suggestion
+ The data replication task in TiCDC is paused for a long time, during which a large amount of incremental data is accumulated and needs to be replicated.
```
Sorry my fault.

> + The data changes subscription task is started from a early time point with a high volume of data writes, thus accumulating a large amount of update data that need to be replicated.

Reviewer 2: ```suggestion
+ The data replication task is started from an early timestamp so it becomes necessary to replication a large amount of incremental data.
```

Author: ```suggestion
+ The data replication task is started from an early timestamp so it becomes necessary to replicate a large amount of incremental data.
```

> **Note:**
>

> > + Use the unified sorter with caution if your servers use mechanical hard drives or other storage devices that have latency or bottlenecks in throughput.

Reviewer: ```suggestion
> + If your servers use mechanical hard drives or other storage devices that have latency or bottlenecks in throughput, use the unified sorter with caution.
```

> > + If your servers use mechanical hard drives or other storage devices that have latency or bottlenecks in throughput, use the unified sorter with caution.

Reviewer 2: ```suggestion
> + If your servers use mechanical hard drives or other storage devices that have high latency or limited bandwidth, use the unified sorter with caution.
```

> > + The total free capacity of hard drives must be greater than or equal to 128G. If you need to replicate a large amount of historical data, make sure that the free capacity of each node is greater than or equal to the size of incremental data that needs to be replicated.

Reviewer 2: ```suggestion
> + The total free capacity of hard drives must be greater than or equal to 128G. If you need to replicate a large amount of historical data, make sure that the free capacity on each node is greater than or equal to the size of the incremental data that needs to be replicated.
```

> > + The total free capacity of hard drives must be greater than or equal to 128G. If you need to replicate a large amount of historical data, make sure that the free capacity of each node is greater than or equal to the the size of incremental data that needs to be replicated.

Reviewer: ```suggestion
> + The total free capacity of hard drives must be greater than or equal to 128G. If you need to replicate a large amount of historical data, make sure that the free capacity of each node is greater than or equal to the size of incremental data that needs to be replicated.
```

> > + Unified sorter is enabled by default. If your servers do not match the above requirements and want to disenable the unified sorter, you need to manually set `sort-engine` to `memory` for the changefeed.

Reviewer: ```suggestion
> + Unified sorter is enabled by default. If your servers do not match the above requirements and you want to disable the unified sorter, you need to manually set `sort-engine` to `memory` for the changefeed.
```
